.github/workflows/general-ci.yml

```diff
@@ -18,3 +18,25 @@ jobs:
 
       - name: Verify formatting
         run: npm run format-check
+
+  dist-up-to-date:
+    name: Built action is up-to-date
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - run: npm install
+
+      - run: npm run package
+
+      - name: Check that repo contents have not been changed by rebuilding the actions
+        run: |
```

Comment:
This is basically to enforce someone calls build before trying to "use"/release/whatever?

Reply:
Yep. It's mainly to enforce that the build output in `dist/index.js` matches the source code. The action itself calls the code in `dist/index.js`, so without a check it's possible that the action uses something that doesn't match the source.

See the other comment for why I think we should commit `dist/` despite these issues.